### PR TITLE
Patch 3

### PR DESCRIPTION
**Pull request by DEntis-T**
Limiter errors

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,2 +1,46 @@
 *.pwn linguist-language=Pawn
 *.inc linguist-language=Pawn
+
```
Comment (Y-Less) [inline .gitattributes:3]: Why is this file in this PR?

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+88""""""""88  a8"     "8a  a8"     "8a  88 ,a8"    I8[    ""  
+88        88  8b       d8  8b       d8  8888[       `"Y8ba,   
+88        88  "8a,   ,a8"  "8a,   ,a8"  88`"Yba,   aa    ]8I  
+88        88   `"YbbdP"'    `"YbbdP"'   88   `Y8a  `"YbbdP"'  
+                                                              
+                                                              
+
+*/
+
+#define YSI_INTERIORS_TRUE 1
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:19]: Why are these defines needed?  Could the API not be simplified to just accept `true`/`false`?

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+#define YSI_INTERIORS_FALSE 0
+
+#define YSI_INTERIOR_ACTION_ENTER 0
+#define YSI_INTERIOR_ACTION_EXIT 1
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:23]: Same here.

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+YSI_OnPlayerNearEntrance(playerid);
+public YSI_OnPlayerNearEntrance(playerid)
+{
+	new YSI_InteriorFunctionName[100];
+	new YSI_InteriorActionid;
+    TogglePlayerControllable(playerid,false);
+	for(new i; i < MAX_INTERIORS; i++)
+	{
+		if( IsPlayerInRangeOfPoint( playerid, 1.5, gYSI_InteriorData[ i ][e_YSI_INTENTRANCE_X], \
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:38]: Please fix the spacing on these brackets.  `if (X)`, not `if( X )`.

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+{
+	TogglePlayerControllable(playerid,true);
+}
+
+YSI_OnPlayerNearEntrance(playerid);
+public YSI_OnPlayerNearEntrance(playerid)
+{
+	new YSI_InteriorFunctionName[100];
+	new YSI_InteriorActionid;
+    TogglePlayerControllable(playerid,false);
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:35]: Tabs, not spaces.

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+	{
+		if( IsPlayerInRangeOfPoint( playerid, 1.5, gYSI_InteriorData[ i ][e_YSI_INTENTRANCE_X], \
+		gYSI_InteriorData[ i ][e_YSI_INTENTRANCE_Y], gYSI_InteriorData[ i ][e_YSI_INTENTRANCE_Z]) \
+		&& GetPlayerVirtualWorld( playerid ) == gYSI_InteriorData[ i ][e_YSI_INTERIOR_VWID_OUT] \
+		&& !IsPlayerInAnyVehicle( playerid ))
+		{
+			YSI_InteriorActionid = 0;
+			format(YSI_InteriorFunctionName,sizeof(YSI_InteriorFunctionName),
+				"%s@_yI",gYSI_InteriorData[ i ][e_YSI_INTERIOR_NAME]);
+
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:46]: The commits pushed have `\r\n`, if you're using git please make sure you only push `\n`, especially if you add a `.gitattributes` saying as much.

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+			if(funcidx(YSI_InteriorFunctionName) != -1) CallLocalFunction(YSI_InteriorFunctionName,"ii",playerid,YSI_InteriorActionid);
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:47]: Missing braces on many many `if`s.

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+
+			SetPlayerInterior( playerid, gYSI_InteriorData[ i ][e_YSI_INTERIOR_INTID] );
+			SetPlayerVirtualWorld( playerid, gYSI_InteriorData[ i ][e_YSI_INTERIOR_VWID] );
+			
+			SetPlayerPos( playerid, gYSI_InteriorData[ i ][e_YSI_INTEXIT_X], gYSI_InteriorData[ i ][e_YSI_INTEXIT_Y], \
+			gYSI_InteriorData[ i ][e_YSI_INTEXIT_Z] );
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:53]: Same here, the spacing is wrong.

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+88  `8b   d8'  88  ""     `Y8  88  88P'   `"8a        d8YaaaaY8b     88""""""'    88  
+88   `8b d8'   88  ,adPPPPP88  88  88       88       d8""""""""8b    88           88  
+88    `888'    88  88,    ,88  88  88       88      d8'        `8b   88           88  
+88     `8'     88  `"8bbdP"Y8  88  88       88     d8'          `8b  88           88  
+                                                                                      
+                                                                                      
+
+*/
+
+stock Interiors_Create3DLabelAndPickup(
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:148]: No `Destroy` function?

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+
+	Interiors_Create3DLabelAndPickup(YSI_EntranceLabelString, YSI_xEntrance,YSI_yEntrance,YSI_zEntrance, -1, -1, 19133, 2.0);
+
+	Interiors_Create3DLabelAndPickup("Exit here!", YSI_xExit,YSI_yExit,YSI_zExit, -1, -1, 19133, 2.0);
+	return YSI_INTERIORS_TRUE;
+}
+
+stock bool:Interiors_IsPlayerInInterior(playerid)
+{
+	if(gYSI_PlayersInterior[ playerid ] = -1) return false;
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:248]: Just return the check directly.

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+	else return true;
+}
+
+stock Interiors_ClearPlayerInteriorData(playerid)
+{
+	gYSI_PlayersInterior[playerid] = -1;
+	return YSI_INTERIORS_TRUE;
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:255]: Why need a return?

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 12]
```diff
@@ -0,0 +1,300 @@
...
+{
+	if(pYSI_Interiorid < -1)
+		return YSI_INTERIORS_FALSE;
+
+	gYSI_PlayersInterior[YSI_p_Playerid] = pYSI_Interiorid;
+	return YSI_INTERIORS_TRUE;
+}
+
+#define Interiors_InteriorActionPerformed(%0) \
+	(YSI_Actionid == %0)
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:300]: This is very poor practice.  This call will only work in some functions, for no apparent reason.  Don't hide variables like this.

File: YSI_Visual/y_interiors/y_interiors_impl.inc [hunk 13]
```diff
@@ -0,0 +1,300 @@
...
+*/
+
+#define interiorcall%0(%1) \
+	forward%0@_yI(%1,YSI_Actionid); \
+	public%0@_yI(%1,YSI_Actionid)
+
+#define hookinteriorcall%0(%1) \
+	hook%0@_yI(%1,YSI_Actionid)
+
+stock Interiors_SetPlayerInterior(YSI_p_Playerid, pYSI_Interiorid)
```
Comment (Y-Less) [inline YSI_Visual/y_interiors/y_interiors_impl.inc:290]: Why all the weird variable names?

**Review by Y-Less — COMMENTED** [review]
I don't know what this is meant to do.  y_unique already checks that values expire.  Also, there's loads of irrelevant things in here.

**Comment by DEntis-T** [comment]
This include is implemented into DETUtils - also thanks for the replies though.

**Pull request closed without merging**